src/Tools/babylon.assetsManager.ts

```diff
@@ -28,8 +28,8 @@ module BABYLON {
             this.taskState = AssetTaskState.INIT;
         }
 
-        onSuccess: (task: this) => void;
-        onError: (task: this, message?: string, exception?: any) => void;
+        onSuccess: (task: AbstractAssetTask) => void;

```

Comment:
This is exactly what i was trying to avoid. Now the task returned from the onSuccess callback will be an abstract asset task and no the real task. This forces the user to cast the task (and we will need to change the documentation :) )

This is the reason it took me long to enable this flag. This requires a small architectural change.